editor/store/effects/shared-blocks.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * External dependencies
+ */
+import { castArray, map, uniqueId } from 'lodash';
+import { BEGIN, COMMIT, REVERT } from 'redux-optimist';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import {
+	parse,
+	serialize,
+	createBlock,
+	isSharedBlock,
+} from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { resolveSelector } from './utils';
+import {
+	receiveSharedBlocks as receiveSharedBlocksAction,
+	createSuccessNotice,
+	createErrorNotice,
+	removeBlocks,
+	replaceBlock,
+	receiveBlocks,
+	saveSharedBlock,
+} from '../actions';
+import {
+	getSharedBlock,
+	getBlock,
+	getBlocks,
+} from '../selectors';
+
+/**
+ * Module Constants
+ */
+const SHARED_BLOCK_NOTICE_ID = 'SHARED_BLOCK_NOTICE_ID';
+
+/**
+ * Fetch Shared Blocks Effect Handler.
+ *
+ * @param {Object} action  action object.
+ * @param {Object} store   Redux Store.
+ */
+export const fetchSharedBlocks = async ( action, store ) => {
+	const { id } = action;
+	const { dispatch } = store;
+
+	// TODO: these are potentially undefined, this fix is in place
+	// until there is a filter to not use shared blocks if undefined
+	const postType = await resolveSelector( 'core', 'getPostType', 'wp_block' );
+	if ( ! postType ) {
```

Comment:
I'm not sure how to check the return value for this selector. We should ensure it is not an empty object because this check will fail.

Just in case it's necessary, there are 2 more places where we have the same check.

Reply:
I don't think it can be an empty object, if the state is not there for some reason, it will be undefined and it will be the right post type if not. I wonder if we can't just remove the check entirely actually :P